Gracefully handle rate limit response when attempting to Sign in with QR

hughns opened the pull request:
<!-- Thanks for submitting a PR! Please ensure the following requirements are met in order for us to review your PR -->

This PR adds handling of a HTTP 429 rate limited response when attempting to get an `m.login.token`.

Also:

- preparation for rust crypto by using `getCrypto()` instead of deprecated `crypto` field
- clean up state after flow completes successfully

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

<!--
If you would like to specify text for the changelog entry other than your PR title, add the following:

Notes: Add super cool feature

Changes in this project also generate changelogs in Element Web. To disable this, use the following:

element-web notes: none

or specify alternative text:

element-web notes: Add super cool feature

Changes to this project require tagging with the type of change. If you know the correct type, add the following:

Type: [enhancement/defect/task]

-->


<!-- CHANGELOG_PREVIEW_START -->
---
This change is marked as an *internal change* (Task), so will not be included in the changelog.<!-- CHANGELOG_PREVIEW_END -->

t3chguy commented on src/components/views/auth/LoginWithQR.tsx:
```diff
@@ -136,16 +142,27 @@ export default class LoginWithQR extends React.Component<IProps, IState> {
+            }
             this.props.onFinished(true);
-        } catch (e) {
+        } catch (e: any) {
             logger.error("Error whilst approving sign in", e);
+            if (e?.httpStatus === 429) {
```
This isn't quite right, `throw "foo"` would cause this to explode for example

```suggestion
        } catch (e) {
            logger.error("Error whilst approving sign in", e);
            if (e instanceof HTTPError && e.httpStatus === 429) {
```

t3chguy commented on src/components/views/auth/LoginWithQR.tsx:
```diff
@@ -136,16 +142,27 @@ export default class LoginWithQR extends React.Component<IProps, IState> {
             this.props.onFinished(true);
         } catch (e) {
             logger.error("Error whilst approving sign in", e);
+            if (e instanceof HTTPError && e.httpStatus === 429) {
```
You'll need to import it from the matrix-js-sdk

hughns replied:
Sorry, I blindly applied the suggested change without testing. My bad.

t3chguy commented:
Please avoid force pushing once you have received review as it breaks any ability to review changes since last review.

hughns commented:
> Please avoid force pushing once you have received review as it breaks any ability to review changes since last review.

Sorry. I will make a mental note to not do this.
